org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/model/PresetRepositoryFactory.java

```diff
@@ -45,7 +45,17 @@
 	private static final File PRESET_STORAGE_DIR = AgentPlugin.getDefault().getStateLocation().append(".presets") // $NON-NLS-1$
 			.toFile();
 
-	public static PresetRepository create() {
+	private static PresetRepository singleton;
+
+	public static PresetRepository createSingleton() {
+		if (singleton == null) {
+			singleton = create();
+		}
+
+		return create();
```

Comment:
Should this be `return singleton`?

Reply:
How can I messed this up... sorry. Fixed in fb253f5